app/assets/javascripts/school_administrator_dashboard/dashboard_components/discipline_dashboard/SchoolDisciplineDashboard.js

```diff
@@ -0,0 +1,184 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import DashboardHelpers from '../DashboardHelpers';
+import StudentsTable from '../StudentsTable';
+import DashboardBarChart from '../DashboardBarChart';
+import DateSlider from '../DateSlider';
+
+class SchoolDisciplineDashboard extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      startDate: DashboardHelpers.schoolYearStart(),
+      selectedChart: {
+        type: "location",
+        data: this.props.disciplineIncidentsByLocation,
+        title: "Incidents by Location"},
+      selectedCategory: null
+    };
+    this.setDate = (range) => {
```

Comment:
Can you say more about what all these methods in the constructor are doing?  If there's some initial state that the component should have, can we just set that in `this.state = {...}` and let render react to that and render the correct things?

Reply:
@kevinrobinson this was just to bind the methods since they're all passed as callbacks. Looking around I guess it's more typical to make a prototype and bind it in the constructor, so I'll switch to that

Reply:
Ah gotcha, I misread this.  Yeah, you can define these as methods just like `filterIncidentDates`, and then do the binding in the constructor, which I think is the same thing your'e saying here :)